fluentd/vendored_gem_src/syslog_protocol/lib/syslog_protocol/packet.rb

```diff
@@ -1,17 +1,34 @@
 module SyslogProtocol
   class Packet
-    attr_reader :facility, :severity, :hostname, :tag
+    attr_reader :facility, :severity, :hostname, :tag, :rfc, :appname, :procid, :msgid
     attr_accessor :time, :content
 
+    NULLVALUE = '-'
+
+    def initialize
+      @rfc = :rfc3164
+      @procid = NULLVALUE
+      @msgid = NULLVALUE
+    end
+
     def to_s
       assemble
     end
 
     def assemble(max_size = 1024)
+      if @rfc == :rfc5424
+        @appname = @tag
+        assemble_rfc5424
```

Comment:
Question: should the default if no RFC is specified be the newer or the older RFC? Is backwards compat more important or facilitating more modern development? I don't know the answer - maybe Christian has thoughts.

Reply:
as of now i have kept it to be older rfc, else we need changes in earlier plugin as well. just wanted to keep that path unchanged.